GetStream/stream-chat-swift
# Implement user block feature

### 🔗 Issue Links

Resolve https://github.com/GetStream/ios-issues-tracking/issues/788

### 🎯 Goal

Implement user blocking feature.

### 🧪 Manual Testing Notes

1. On first entry
- Enable blocking feature in configuration
- Open channel long-press participant's message
- Verify action item (`Block user` / `Unblock user`)

2. On reconnect
- Enable blocking feature in configuration
- Open channel
- Block user
- Log out and log in as the same user
- Open channel long-press participant's message
- Verify action item (`Block user` / `Unblock user`)

3. Get the list of blocked users
- Enable blocking feature in configuration
- Open channel
- Block user
- Open channel debug menu
- Open blocked users view
- Verify that blocked user is in the list

4. 1:1 channel
- Enable blocking feature in configuration
- Open a 1:1 channel
- Block user
- Send a message as a blocked user
- There should be no messages from the blocked user

5. Group channel
- Enable blocking feature in configuration
- Open a group channel
- Block some user
- Send a message as a blocked user
- The message should be visible to the main user

6. The visibility of the 1:1 channel with a blocked user
- Enable blocking feature in configuration
- Open a 1:1 channel
- Block user
- Go back to channel list
- The 1:1 channel with the blocked user should be hidden
- Open a group channel
- Unblock that user
- Go back to channel list
- The 1:1 channel with the unblocked user should be visible

### ☑️ Contributor Checklist


### 🎁 Meme

![gif](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExN2hldnBuZWFmcTBqYnEzNXU0N2hoczF5ajF2Z3ZnODk5ZWUybTdjbyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/5kOmeG2xGGARurUF3g/giphy.gif)
